derp/derp_server.go

```diff
@@ -147,8 +157,8 @@ func (s *Server) registerClient(c *sclient) {
 		c.logf("adding connection")
 	} else {
 		s.clientsReplaced.Add(1)
-		old.nc.Close()
-		c.logf("adding connection, replacing %s", old.nc.RemoteAddr())
+		go old.nc.Close()
+		c.logf("adding connection, replacing %s", old.remoteAddr)
```

Comment:
swap these lines so we see this in the logs before any sort of activity/logging related to the nc.Close happening later.